### PR TITLE
feat: Store game path in persistent store

### PR DESCRIPTION
**Pull request by GeckoEidechse**
WIP commit still the whole logic is a bit clunky, might wanna refactor

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
In particular we probably just wanna store path always, not just when setting custom one. For that I will also need to store type, not just path.

**Comment by GeckoEidechse** [comment]
Storing custom paths should work now. Just not sure if I want to merge this before I have a way of nuking persistent store in the face of errors...

File: src-vue/src/plugins/store.ts [hunk 1]
```diff
@@ -226,21 +234,51 @@ async function _initializeApp(state: any) {
...
+                    duration: 0 // Duration `0` means the notification will not auto-vanish
+                });
             });
-        });
-    const typedResult: GameInstall = result as GameInstall;
-    state.game_path = typedResult.game_path;
-    state.install_type = typedResult.install_type;
+    }
+
+    if (result !== undefined) { // Found some form of value for gameinstall
```
Comment (GeckoEidechse) [inline src-vue/src/plugins/store.ts:271]: This check catches an exception that currently happens on `main` when no valid gamepath was found.

**Comment by Alystrasz** [comment]
When opening the app for the first time (i.e. without selecting a path), the button displays "Install" instead of "Select a game folder".

image

File: src-vue/src/plugins/store.ts [hunk 2]
```diff
@@ -226,21 +234,51 @@ async function _initializeApp(state: any) {
...
-            console.error(err);
-            notification_handle = ElNotification({
-                title: 'Titanfall2 not found!',
-                message: "Please manually select install location",
-                type: 'error',
-                position: 'bottom-right',
-                duration: 0 // Duration `0` means the notification will not auto-vanish
+    var result = undefined;
+    var persistent_game_install = (await persistentStore.get('game-install')) as any;
+    console.log(persistent_game_install);
```
Comment (Alystrasz) [inline src-vue/src/plugins/store.ts:239]: Don't forget to remove this :)

**Review by Alystrasz — CHANGES_REQUESTED** [review]

**Comment by GeckoEidechse** [comment]
> When opening the app for the first time (i.e. without selecting a path), the button displays "Install" instead of "Select a game folder".

Hmm, weird. Does it have a path set in settings?

**Comment by GeckoEidechse** [comment]
> > When opening the app for the first time (i.e. without selecting a path), the button displays "Install" instead of "Select a game folder".
> 
> Hmm, weird. Does it have a path set in settings?

Ok, can reproduce, checking what's causing it now ^^

**Comment by GeckoEidechse** [comment]
> > > When opening the app for the first time (i.e. without selecting a path), the button displays "Install" instead of "Select a game folder".
> > 
> > 
> > Hmm, weird. Does it have a path set in settings?
> 
> Ok, can reproduce, checking what's causing it now ^^

Fixed with 66101c22128ce81618d209f324f2d56873b21b17

File: src-vue/src/plugins/store.ts (hunk 2 above)
Comment (GeckoEidechse) [reply src-vue/src/plugins/store.ts]: Removed with aaf142767ea36bdbc87345612e64ea6610999190 ^^

**Review by Alystrasz — APPROVED** [review]

**Pull request merged**